system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -253,6 +266,28 @@ this.TopStoriesFeed = class TopStoriesFeed {
     this.newTabsSinceSpoc++;
   }
 
+  /**
+   * Update bookmark title, description and image with metadata from Pocket.
+   * @param url Used to identify if the bookmark was a topstory.
+   */
+  updateBookmarkMetadata(url) {
+    if (!this.stories) {
+      return;
+    }
+
+    const story = this.stories.find(s => s.url === url);
```

Comment:
Why not just include the data you need here in the event from content? Isn't the data the most reliable / guaranteed to be there at that point?

Reply:
@k88hudson the bookmark moves into highlights after refresh, not when you bookmark. So there still needs to be some exchange from Highlights to TopStories to get the bookmarks.

Reply:
I think the tricky case is on fresh browser start. When the App has no state.